pkg/gatherers/clusterconfig/gather_lokistack.go

```diff
@@ -0,0 +1,89 @@
+package clusterconfig
+
+// nolint: dupl
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog/v2"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+
+	"github.com/openshift/insights-operator/pkg/record"
+	"github.com/openshift/insights-operator/pkg/utils"
+	"github.com/openshift/insights-operator/pkg/utils/anonymize"
+)
+
+// GatherOpenshiftLogging Collects `clusterlogging.logging.openshift.io` resources.
+//
+// ### API Reference
+// - https://github.com/openshift/cluster-logging-operator/blob/master/pkg/apis/logging/v1/clusterlogging_types.go
+//
+// ### Sample data
+// - docs/insights-archive-sample/config/logging/openshift-logging/instance.json
+//
+// ### Location in archive
+// - `config/logging/{namespace}/{name}.json`
+//
+// ### Config ID
+// `clusterconfig/openshift_logging`
+//
+// ### Released version
+// - 4.9.0
+//
+// ### Backported versions
+// None
+//
+// ### Changes
+// None
+func (g *Gatherer) GatherLokiStack(ctx context.Context) ([]record.Record, []error) {
+	gatherDynamicClient, err := dynamic.NewForConfig(g.gatherKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	return gatherLokiStack(ctx, gatherDynamicClient)
+}
+
+func gatherLokiStack(ctx context.Context, dynamicClient dynamic.Interface) ([]record.Record, []error) {
+	loggingResourceList, err := dynamicClient.Resource(lokiStackResource).List(ctx, metav1.ListOptions{})
+	if errors.IsNotFound(err) {
+		return nil, nil
+	}
+	if err != nil {
+		klog.V(2).Infof("Unable to list %s resource due to: %s", lokiStackResource, err)
+		return nil, []error{err}
+	}
+
+	var records []record.Record
+	for i := range loggingResourceList.Items {
+		item := loggingResourceList.Items[i]
+		anonymizeLokiStackNamespace(item.Object)
+		records = append(records, record.Record{
+			Name: fmt.Sprintf("config/lokistack/%s/%s", item.GetUID(), item.GetName()),
```

Comment:
I would put this to the `namespace` path similar as https://github.com/openshift/insights-operator/blob/master/pkg/gatherers/clusterconfig/gather_openstackversions.go#L56

Reply:
Right now we are not using the namespace, but the UID of the CR, what should be, in your opinion, the right path for the files?

Reply:
I think it's safer to use UIDs, but I would ask in the ticket for opinions.

Reply:
This topic was already discussed and closed. No UID will be used, as only openshift-logging resources will be kept on the archive, and there is no need to anonymize that one